File: src/main/java/com/lessmarkup/dataobjects/UserPropertyDefinition.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

public class UserPropertyDefinition extends AbstractDataObject {
    private String name;
    private String title;
    private UserPropertyType type;
    private boolean required;

    @RequiredField
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public String getTitle() {
        return title;
    }
    public void setTitle(String title) {
        this.title = title;
    }

    @RequiredField
    public UserPropertyType getType() {
        return type;
    }
    public void setType(UserPropertyType type) {
        this.type = type;
    }

    public boolean isRequired() {
        return required;
    }
    public void setRequired(boolean required) {
        this.required = required;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/UserPropertyDefinition.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import com.lessmarkup.dataobjects.UserPropertyType.UserPropertyType
import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.DataObject

class UserPropertyDefinition extends DataObject {
  @RequiredField
  var name: String = null
  var title: String = null
  @RequiredField
  var propertyType: UserPropertyType = UserPropertyType.NOTE
  var required: Boolean = false
}
